pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -0,0 +1,123 @@
+package monitoringstack
+
+import (
+	"fmt"
+
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/rhobs/observability-operator/pkg/apis/monitoring/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	monitoringStackAvailable       = "MonitoringStackAvailable"
+	monitoringStackReconciled      = "MonitoringStackReconciled"
+	prometheusNotAvailable         = "PrometheusNotAvailable"
+	prometheusNotReconciled        = "PrometheusNotReconciled"
+	cannotReadPrometheusConditions = "Cannot read Prometheus status conditions"
+)
+
+func updateConditions(msConditions []v1alpha1.Condition, promConditions []monv1.PrometheusCondition, generation int64, recError error) []v1alpha1.Condition {
```

Comment:
Thank you for changing the implementation. 
Could you please consider adding a unit test for different scenarios?

Reply:
OK. Good point.